lib/absinthe/schema/notation.ex

```diff
@@ -879,16 +892,65 @@ defmodule Absinthe.Schema.Notation do
       _, node ->
         node
     end
+  end
+  ```
+
+  ## Applying a type system directive
+  Directives can be applied in your schema. E.g. by default the `@deprecated`
+  directive is available to be applied to fields and enum values.
+
+  You can define your own type system directives. See `Absinthe.Schema.Prototype`
+  for more information.
+
+  ### Placement
+
+  #{Utils.placement_docs(@placement, :applied_directive)}
 
+  ### Examples
+
+  When you have a type system directive named `:feature` it can be applied as
+  follows:
+
+  ```elixir
+  object :post do
+    directive :feature, name: ":object"
+
+    field :name, :string do
+      deprecate "Bye"
+    end
+  end
+
+  scalar :sweet_scalar do
+    directive :feature, name: ":scalar"
+    parse &Function.identity/1
+    serialize &Function.identity/1
   end
   ```
   """
-  defmacro directive(identifier, attrs \\ [], do: block) do
+  defmacro directive(identifier, attrs, do: block) when is_list(attrs) when not is_nil(block) do
```

Comment:
`directive/1,2,3` can be used for both defining directives as applying them to types in the schema. 

I think this is clear for the users but perhaps there's another function name we can use for applying the directive.